core/src/main/java/ai/timefold/solver/core/config/util/ConfigUtils.java

```diff
@@ -415,6 +425,7 @@ Maybe the member (%s) should return a parameterized %s."""
                                 memberName, type, memberName, type.getSimpleName())));
     }
 
+    // TODO
     public static Optional<Class<?>> extractGenericTypeParameter(String parentClassConcept, Class<?> parentClass, Class<?> type,
             Type genericType, Class<? extends Annotation> annotationClass, String memberName) {
```

Comment:
@triceo I suppose the parameters all should be `@NonNull`, right?

Reply:
Just wanted to change this accordingly and found there is a null-check for `annotationClass`, so I'll make that param nullable